src/main/java/pl/edu/zut/mad/schedule/ErrorHandler.java

```diff
@@ -26,33 +27,50 @@ public ErrorHandler(MessageSource messageSource) {
         this.messageSource = messageSource;
     }
 
-    @ExceptionHandler(NotFoundException.class)
-    @ResponseStatus(value = HttpStatus.NOT_FOUND)
-    @ResponseBody
-    public ErrorMessage handleNotFoundException(NotFoundException ex) {
+    @ExceptionHandler({
+            NotFoundException.class,
+            EmptyDatabaseException.class,
+            RuntimeException.class
+    })
+    public ResponseEntity<ErrorMessage> handleException(Exception ex, HttpServletRequest request) {
+        if (ex instanceof NotFoundException) {
+            return handleNotFoundException((NotFoundException) ex, request);
+        } else if (ex instanceof EmptyDatabaseException) {
+            return handleEmptyDatabaseException((EmptyDatabaseException) ex, request);
+        } else {
+            return handleRuntimeException((RuntimeException) ex, request);
+        }
+    }
```

Comment:
In future we can think about soulution with polimorphizm, this could reduce `if` checking.